src/main/scala/is/hail/methods/LDMatrix.scala

```diff
@@ -18,31 +18,26 @@ object LDMatrix {
     */
   def apply(vds : VariantDataset, optComputeLocally: Option[Boolean]): LDMatrix = {
     val nSamples = vds.nSamples
-    val originalVariants = vds.variants.collect()
-    assert(originalVariants.isSorted, "Array of variants is not sorted. This is a bug.")
+    val nVariants = vds.countVariants()
 
-    val normalizedIRM = ToNormalizedIndexedRowMatrix(vds)
-    normalizedIRM.rows.persist()
-    val variantKeptIndices = normalizedIRM.rows.map{case IndexedRow(idx, _) => idx.toInt}.collect()
-    assert(variantKeptIndices.isSorted, "Array of kept variants is not sorted. This is a bug.")
-    val variantsKept = variantKeptIndices.map(idx => originalVariants(idx))
+    val normalizedHardCalls = vds.rdd.map { case (v, (va, gs)) => (v, RegressionUtils.normalizedHardCalls(gs, nSamples))}
+    val filteredNormalizedHardCalls = normalizedHardCalls.filter{ case (v, opt) => !opt.isEmpty}
```

Comment:
These two lines could be:

```scala
val filteredNormalizedHardCalls = vds.rdd.flatMap { 
  case (v, (va, gs)) => RegressionUtils.normalizedHardCalls(gs, nSamples).map(x => (v, x))
}
```

right? I don't think the performance will be much different and it seems a bit easier to read.